bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceDialog.java

```diff
@@ -898,35 +898,58 @@ private void storeSettings() {
 		writeConfiguration();
 	}
 
+	/**
+	 * Initializes the string to search for and the according text in the find field
+	 * based on either the selection in the target or, if nothing is selected, with
+	 * the newest search history entry.
+	 */
+	private void initFindString() {
+		if (!okToUse(fFindField)) {
+			return;
+		}
+
+		fFindField.removeModifyListener(fFindModifyListener);
+		if (!initFindStringFromSelection()) {
+			initFindStringFromHistory();
+		}
+		fFindField.setSelection(new Point(0, fFindField.getText().length()));
+		fFindField.addModifyListener(fFindModifyListener);
+	}
+
 	/**
 	 * Initializes the string to search for and the appropriate text in the Find
 	 * field based on the selection found in the action's target.
 	 */
-	private void initFindStringFromSelection() {
+	private boolean initFindStringFromSelection() {
```

Comment:
I wonder if adding a return type to this function might not be giving too much responsibility to the function?

Reply:
Coupling the query logic with action logic in one method is not necessary here. I have cleaned the code by separating into two methods in 033648c.